ffi/src/scan.rs

```diff
@@ -83,27 +100,34 @@ unsafe fn get_raw_arrow_data_impl(
     Ok(Box::leak(ret_data))
 }
 
-impl BoxHandle for Scan {}
+pub struct ScanHandle {
```

Comment:
Probably should name this `SharedScan` so we end up with a `HandleSharedScan` on the c side and not a `HandleScanHandle`

Reply:
Actually it's not shared I guess, so either `MutScan` or `ExclusiveScan`?

Reply:
Yeah scan is a weird one (duckdb was hitting concurrency bugs here). It has to be mutable, because consuming from the scan modifies its internal state. I think we fixed the bug by adding a mutex internally, but technically duckdb should probably be doing the mutex work to ensure there's only one mutable reference possible at a time??